token/link/link.go

```diff
@@ -98,12 +99,20 @@ func (service *LinkService) ProviderLocation(ctx context.Context, req *goa.Reque
 		return "", errs.NewBadParameterError("redirect", redirectURL).Expected("valid URL")
 	}
 	parameters := linkURL.Query()
-	parameters.Add(identityIDParam, identityID)
-	parameters.Add(forParam, forResource)
+	parameters.Set(identityIDParam, identityID)
+
+	// If "for" contains multiple resources then do linking one by one
+	forResources := strings.Split(forResource, ",")
+	if len(forResources) > 1 {
+		parameters.Set(nextParam, strings.Join(forResources[1:], ","))
+	} else {
+		parameters.Del(nextParam)
```

Comment:
This plays a role when the second( and last ) linking is being done?

Reply:
Correct.